### ISSUE TITLE
Crash sometimes when dragging with stylus

### ISSUE THREAD

**Issue by piegamesde** [xournalpp/xournalpp#2470]
**Affects versions :**
 - OS: Arch Linux
 - (Linux only) Desktop environment: Gnome 3.38 + Wayland
 - `libgtk: 3.24.21`
 - `Xournal++ 1.0.19`
 - Installation method: Nixpkgs 20.09 release channel

**Describe the bug**
Sometimes when I drag the pane with the stylus it crashes

**To Reproduce**
Didn't find anything more deterministic yet. Happens roughly once or twice an hour.

**Application log**

```
** (com.github.xournalpp.xournalpp:266938): WARNING **: 19:18:13.429: Input for the same device started twice!
** Message: 19:18:40.351: Info: autosave document...
** Message: 19:18:40.370: Autosave renamed
** Message: 19:18:40.385: Autosaving

** (com.github.xournalpp.xournalpp:266938): WARNING **: 22:42:05.434: XojTypeCheck failed: expected NewGtkInputDevice but get something else on /build/source/src/gui/inputdevices/old/NewGtkInputDevice.cpp:64

** (com.github.xournalpp.xournalpp:266938): WARNING **: 22:42:05.434: XojTypeCheck failed: expected NewGtkInputDevice but get Invalid_type on /build/source/src/gui/inputdevices/old/NewGtkInputDevice.cpp:64

** (com.github.xournalpp.xournalpp:266938): WARNING **: 22:42:05.434: [Crash Handler] Crashed with signal 11

** (com.github.xournalpp.xournalpp:266938): WARNING **: 22:42:05.434: [Crash Handler] Wrote crash log to: ~/.xournalpp/errorlogs/errorlog.20201130-224205.log

addr2line: DWARF error: could not find variable specification at offset 20e4
addr2line: DWARF error: could not find variable specification at offset 2140
addr2line: DWARF error: could not find variable specification at offset 219c
addr2line: DWARF error: could not find variable specification at offset 339e
addr2line: DWARF error: could not find variable specification at offset 33aa
[…]
addr2line: DWARF error: could not find variable specification at offset 3b28
```

**Crash log**

```
Date: Mon Nov 30 22:42:05 2020
Error: signal 11
[bt]: (0) ~/.nix-profile/bin/xournalpp() [0x5eb3ea]
[bt]: (1) /nix/store/HASH-glibc-2.31/lib/libc.so.6(+0x380f0) [0x7fa1c3af90f0]
[bt]: (2) /nix/store/HASH-glib-2.64.5/lib/libgobject-2.0.so.0(g_type_check_instance_cast+0x16) [0x7fa1c441df26]
[bt]: (3) ~/.nix-profile/bin/xournalpp() [0x584df0]
[bt]: (4) ~/.nix-profile/bin/xournalpp() [0x5f45ee]
[bt]: (5) /nix/store/HASH-gtk+3-3.24.21/lib/libgdk-3.so.0(+0x31b99) [0x7fa1c493db99]
[bt]: (6) /nix/store/HASH-glib-2.64.5/lib/libglib-2.0.so.0(g_main_context_dispatch+0x14e) [0x7fa1c521331e]
[bt]: (7) /nix/store/HASH-glib-2.64.5/lib/libglib-2.0.so.0(+0x536d0) [0x7fa1c52136d0]
[bt]: (8) /nix/store/HASH-glib-2.64.5/lib/libglib-2.0.so.0(g_main_loop_run+0x83) [0x7fa1c52139a3]
[bt]: (9) /nix/store/HASH-gtk+3-3.24.21/lib/libgtk-3.so.0(gtk_main+0x75) [0x7fa1c4bf2f45]
[bt]: (10) ~/.nix-profile/bin/xournalpp(_ZN11XournalMain3runEiPPc+0xe1c) [0x4cca4c]
[bt]: (11) ~/.nix-profile/bin/xournalpp(main+0x33) [0x49d773]
[bt]: (12) /nix/store/HASH-glibc-2.31/lib/libc.so.6(__libc_start_main+0xed) [0x7fa1c3ae4c7d]
[bt]: (13) ~/.nix-profile/bin/xournalpp(_start+0x2a) [0x49dd0a]


Try to get a better stracktrace...
[bt] #1 ~/.nix-profile/bin/xournalpp() [0x5eb88d]
/build/source/src/util/CrashHandlerUnix.h:102
[bt] #2 /nix/store/HASH-glibc-2.31/lib/libc.so.6(+0x380f0) [0x7fa1c3af90f0]
??:0
[bt] #3 /nix/store/HASH-glib-2.64.5/lib/libgobject-2.0.so.0(g_type_check_instance_cast+0x16) [0x7fa1c441df26]
??:0
[bt] #4 ~/.nix-profile/bin/xournalpp() [0x584df0]
/build/source/src/gui/inputdevices/old/InputSequence.cpp:171
[bt] #5 ~/.nix-profile/bin/xournalpp() [0x5f45ee]
/nix/store/HASH-gcc-9.3.0/include/c++/9.3.0/bits/std_function.h:259
[bt] #6 /nix/store/HASH-gtk+3-3.24.21/lib/libgdk-3.so.0(+0x31b99) [0x7fa1c493db99]
??:0
[bt] #7 /nix/store/HASH-glib-2.64.5/lib/libglib-2.0.so.0(g_main_context_dispatch+0x14e) [0x7fa1c521331e]
??:0
[bt] #8 /nix/store/HASH-glib-2.64.5/lib/libglib-2.0.so.0(+0x536d0) [0x7fa1c52136d0]
??:0
[bt] #9 /nix/store/HASH-glib-2.64.5/lib/libglib-2.0.so.0(g_main_loop_run+0x83) [0x7fa1c52139a3]
??:0
[bt] #10 /nix/store/HASH-gtk+3-3.24.21/lib/libgtk-3.so.0(gtk_main+0x75) [0x7fa1c4bf2f45]
??:0
[bt] #11 ~/.nix-profile/bin/xournalpp(_ZN11XournalMain3runEiPPc+0xe1c) [0x4cca4c]
/build/source/src/control/XournalMain.cpp:438
[bt] #12 ~/.nix-profile/bin/xournalpp(main+0x33) [0x49d773]
/build/source/src/Xournalpp.cpp:43
[bt] #13 /nix/store/HASH-glibc-2.31/lib/libc.so.6(__libc_start_main+0xed) [0x7fa1c3ae4c7d]
??:0
[bt] #14 ~/.nix-profile/bin/xournalpp(_start+0x2a) [0x49dd0a]
/build/glibc-2.31/csu/../sysdeps/x86_64/start.S:122
```

**Comment by rolandlo** [comment]
Have you activated the new input system?

**Comment by piegamesde** [comment]
@rolandlo I thought so, but apparently I haven't.

**Comment by LittleHuba** [comment]
@piegamesde please switch to the new input system and report if the bug persists. That was a known issue of the old system and should already be fixed by the new one.

**Comment by piegamesde** [comment]
I tried out the new input system. No crash so far (I only tried it for five minutes or so), but the new behaviour gives me unusably bad UX, so I'm going to switch back for now.

**Comment by rolandlo** [comment]
What "unusably bad UX" did you experience. I would recommend you to provide a clear description of what you don't like, since the old input system will be gone soon.

**Comment by piegamesde** [comment]
Oh, okay. I looked for some issues to discuss it there but then I didn't find any and forgot about it.

Basically, pressing a stylus button only activates the tool without using it. I still have to press the tip against the tablet for it to do anything. This is an extra unnecessary action, also I created a lot of small dots while changing the tools (releasing the eraser button before lifting the pen again will leave a small dot). Additionally, dragging does not work well for me anymore: it only works sporadically and leaves some dots behind. I suspect some loose contact of my stylus button without proper debouncing to cause this.
